test/DbSQLiteAdapterTest/Tests/DbConnectionTest.cpp

```diff
@@ -39,93 +44,82 @@ namespace systelab { namespace db { namespace sqlite { namespace unit_test {
+		systelab::db::sqlite::ConnectionConfiguration configuration(m_dbFilePath.string());
+		ASSERT_THAT(m_connection.loadDatabase(configuration), NotNull());
 	}
 
 	TEST_F(DbConnectionTest, testLoadDatabaseForHappyPathWithoutEncryptionCreatesDBFile)
 	{
-		std::string dbFilepath = m_dbFilePath.string();
-		systelab::db::sqlite::ConnectionConfiguration configuration(dbFilepath);
-
-		systelab::db::sqlite::Connection connection;
-		auto database = connection.loadDatabase(configuration);
+		systelab::db::sqlite::ConnectionConfiguration configuration(m_dbFilePath.string());
+		m_connection.loadDatabase(configuration);
 		ASSERT_TRUE(std::filesystem::exists(m_dbFilePath));
 	}
 
 
 	// Happy path WITH encryption
 	TEST_F(DbConnectionTest, testLoadDatabaseForHappyPathWithEncryptionReturnsDatabaseObject)
 	{
-		std::string dbFilepath = m_dbFilePath.string();
-		systelab::db::sqlite::ConnectionConfiguration configuration(dbFilepath, "MyEncryptionKey"s);
-
-		systelab::db::sqlite::Connection connection;
-		auto database = connection.loadDatabase(configuration);
-		ASSERT_TRUE(database != NULL);
+		systelab::db::sqlite::ConnectionConfiguration configuration(m_dbFilePath.string(), ENCRYPTION_KEY);
+		ASSERT_THAT(m_connection.loadDatabase(configuration), NotNull());
 	}
 
 	TEST_F(DbConnectionTest, testLoadDatabaseForHappyPathWithEncryptionCreatesDBFile)
 	{
-		std::string dbFilepath = m_dbFilePath.string();
-		systelab::db::sqlite::ConnectionConfiguration configuration(dbFilepath, "MyEncryptionKey"s);
+		systelab::db::sqlite::ConnectionConfiguration configuration(m_dbFilePath.string(), ENCRYPTION_KEY);
+		m_connection.loadDatabase(configuration);
 
-		systelab::db::sqlite::Connection connection;
-		auto database = connection.loadDatabase(configuration);
 		ASSERT_TRUE(std::filesystem::exists(m_dbFilePath));
 	}
 
 	TEST_F(DbConnectionTest, testLoadDatabaseForHappyPathWithEncryptionOverAnExistingDatabase)
 	{
-		std::string dbFilepath = m_dbFilePath.string();
-		std::string encryptionKey = "MyEncryptionKey";
-		systelab::db::sqlite::ConnectionConfiguration configuration(dbFilepath, encryptionKey);
-
-		systelab::db::sqlite::Connection connection;
-		auto database1 = connection.loadDatabase(configuration);
-		createTable(*database1, "MY_TABLE", 10);
-		database1.reset();
+		systelab::db::sqlite::ConnectionConfiguration configuration(m_dbFilePath.string(), ENCRYPTION_KEY);
+		createDatabaseContent(configuration);
 
-		ASSERT_NO_THROW(connection.loadDatabase(configuration));
+		ASSERT_NO_THROW(m_connection.loadDatabase(configuration));
 	}
 
+	TEST_F(DbConnectionTest, testLoadDatabaseAtReadOnlyMode)
+	{
+		systelab::db::sqlite::ConnectionConfiguration validConfiguration(m_dbFilePath.string(), ENCRYPTION_KEY);
+		createDatabaseContent(validConfiguration);
+
+		systelab::db::sqlite::ConnectionConfiguration readOnlyConfiguration(m_dbFilePath.string(), ENCRYPTION_KEY, true);
+		ASSERT_THAT(m_connection.loadDatabase(readOnlyConfiguration), NotNull());
```

Comment:
I'm missing a couple of tests in relation to the read-only mode:
* Perform query (select statements) on a read-only mode database successfully
* Try to perform a modification (insert statement) on a read-only mode and verify it fails (throws?)